ct/homarr.sh

```diff
@@ -54,20 +54,38 @@ function default_settings() {
 
 function update_script() {
 header_info
-if [[ ! -d /opt/homarr ]]; then msg_error "No ${APP} Installation Found!"; exit; fi
-msg_info "Updating $APP (Patience)"
-systemctl stop homarr
-cd /opt/homarr
-if ! git pull; then
-  echo "Already up to date."
+if [[ ! -d /opt/adventurelog ]]; then msg_error "No ${APP} Installation Found!"; exit; fi
+if (( $(df /boot | awk 'NR==2{gsub("%","",$5); print $5}') > 80 )); then
+  read -r -p "Warning: Storage is dangerously low, continue anyway? <y/N> " prompt
+  [[ ${prompt,,} =~ ^(y|yes)$ ]] || exit
+fi
+RELEASE=$(curl -s https://api.github.com/repos/ajnart/homarr/releases/latest | grep "tag_name" | awk '{print substr($2, 3, length($2)-4) }')
+if [[ ! -f /opt/${APP}_version.txt ]] || [[ "${RELEASE}" != "$(cat /opt/${APP}_version.txt)" ]]; then
+  msg_info "Stopping Services"
+  systemctl stop homarr
+  msg_ok "Services Stopped"
+
+  msg_info "Updating ${APP} to ${RELEASE}"
+  cp /opt/homarr/.env /opt/.env
+  wget -q "https://github.com/ajnart/homarr/archive/refs/tags/v${RELEASE}.zip"
+  unzip -q v${RELEASE}.zip
+  mv homarr-${RELEASE} /opt/homarr
```

Comment:
delete old directory first?

Reply:
but should also doublecheck that there is no data/config there that needs to be migrated

Reply:
done